staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -332,7 +332,9 @@ func (rules *ClientConfigLoadingRules) GetExplicitFile() string {
 	return rules.ExplicitPath
 }
 
-// IsDefaultConfig returns true if the provided configuration matches the default
+// IsDefaultConfig returns true if the provided configuration matches the default, but
```

Comment:
When user didn't specify kubeconfig file or  options like "--server", "--token", "--user", kubectl will try to read in-cluster configuration(service account).

After this change, when user specify "--as", "--as-group" or "--timeout" but doesn't specify other auth infos, IsDefaultConfig() still returns true, so that we can try to use in-cluster configuration when --as, --as-group is specified.